src/web/routes.rs

```diff
@@ -10,9 +10,18 @@ pub(super) const DOC_RUST_LANG_ORG_REDIRECTS: &[&str] =
 pub(super) fn build_routes() -> Routes {
     let mut routes = Routes::new();
 
-    routes.static_resource("/robots.txt", super::sitemap::robots_txt_handler);
-    routes.static_resource("/sitemap.xml", super::sitemap::sitemap_handler);
-    routes.static_resource("/opensearch.xml", super::opensearch_xml_handler);
+    routes.static_resource("/robots.txt", PermanentRedirect("/-/static/robots.txt"));
+    routes.static_resource("/favicon.ico", PermanentRedirect("/-/static/favicon.ico"));
+
+    // These should not need to be served from the root as we reference the inner path in links,
+    // but clients might have cached the url and need to update it.
+    routes.static_resource(
+        "/opensearch.xml",
+        PermanentRedirect("/-/static/opensearch.xml"),
+    );
+    routes.static_resource("/sitemap.xml", PermanentRedirect("/-/sitemap.xml"));
```

Comment:
The sitemap has to live at the top level:

> A sitemap can be posted anywhere on your site, but a sitemap affects only descendants of the parent directory. Therefore, a sitemap posted at the site root can affect all files on the site, which is where we recommend posting your sitemaps.

https://support.google.com/webmasters/answer/183668?hl=en

Reply:
Ok, that was one I was wondering about, I'll change this back and put in a comment.